core/domain/event_services.py

```diff
@@ -93,6 +94,57 @@ def _handle_event(
                 rule_spec_index, classification_categorization, params,
                 session_id, time_spent_in_secs))
 
+        is_feedback_useful = True
+        if classification_categorization == (
+                exp_domain.DEFAULT_OUTCOME_CLASSIFICATION):
+            is_feedback_useful = False
+
+        stats_models.AnswerSubmittedEventLogEntryModel.create(
+            exploration_id, exploration_version, state_name, session_id,
+            time_spent_in_secs, is_feedback_useful)
+
+        update_params = {
+            'is_feedback_useful': is_feedback_useful
+        }
+        if feconf.ENABLE_NEW_STATS_FRAMEWORK:
+            stats_services.update_stats(
+                exploration_id, exploration_version, state_name, cls.EVENT_TYPE,
+                update_params)
+
+
+class ExplorationActualStartEventHandler(BaseEventHandler):
+    """Event handler for recording exploration actual start events."""
+
+    EVENT_TYPE = feconf.EVENT_TYPE_ACTUAL_START_EXPLORATION
+
+    @classmethod
+    def _handle_event(
+            cls, exp_id, exp_version, state_name, session_id,
+            client_time_spent_in_secs):
+        stats_models.ExplorationActualStartEventLogEntryModel.create(
+            exp_id, exp_version, state_name, session_id,
+            client_time_spent_in_secs)
+        if feconf.ENABLE_NEW_STATS_FRAMEWORK:
+            stats_services.update_stats(
+                exp_id, exp_version, state_name, cls.EVENT_TYPE, {})
+
+
+class SolutionHitEventHandler(BaseEventHandler):
+    """Event handler for recording solution hit events."""
+
+    EVENT_TYPE = feconf.EVENT_TYPE_SOLUTION
```

Comment:
Look at existing event types: ANSWER_SUBMITTED, DEFAULT_ANSWER_RESOLVED, etc. Let's have all our new events follow the same format: noun_past-tense-verb.

Reply:
Done.